File: ADP.DS.FrontOffice.FI.WebApi/Controllers/ProductsController.cs
using System.Collections.Generic;
using System.Web.Http;
using ADP.DS.FrontOffice.FI.DataManager.Managers;
using ADP.DS.FrontOffice.FI.Model;
using AttributeRouting.Web.Http;

namespace ADP.DS.FrontOffice.FI.Controllers
{
    
    public class ProductsController : ApiController
    {
      
        [GET("api/products/{dealerId}/{cashType}")]
        public IList<ProductTypeDetail> GetProductDetails(int dealerId,string cashType)
        {
            var productDetailManager = new ProductDetailManager();

            return productDetailManager.GetProductDetails(dealerId, bool.Parse(cashType));
        }
       
    }
}

File: ADP.DS.FrontOffice.FI.WebApi/js/Controllers/productsController.js
'use strict';
/* App Controllers */

productsApp.controller('productsController',
    function productsController($scope, productsService, sharedService) {
        //default values for dealerId, cashType
        $scope.cashType = true;
        sharedService.dealerId = 462;

        //get products from products service
        var products = productsService.getDealerProducts(sharedService.dealerId, $scope.cashType);
        products.then(function (data) {
            $scope.dealerProducts = data;
        });

    });